File: Python/Data_Structure_and_Algorithm/Sort/Selection_sort.py
# Selection_sort_v1
def selection_sort_v1(a_list):
    # Iterate through the list
    for i, _ in enumerate(a_list):
        # Assume the current index as the index of the minimum element
        min_idx = i
        # Find the index of the minimum element in the unsorted part of the list
        for j in range(i + 1, len(a_list)):
            if a_list[j] < a_list[min_idx]:
                min_idx = j
        # If the index of the minimum element is not the same as the current index, swap them
        if min_idx != i:
            a_list[min_idx], a_list[i] = a_list[i], a_list[min_idx]

# Selection_sort_v2
def selection_v2(arr):
    # Iterate through the list
    for i in range(len(arr)):
        # Assume the current index as the index of the minimum element
        mini = i
        # Find the index of the minimum element in the unsorted part of the list
        for j in range(i, len(arr)):
            if arr[mini] > arr[j]:
                mini = j
        # If the index of the minimum element is not equal to i, swap them
        if mini != i:
            arr[mini], arr[i] = arr[i], arr[mini]

File: C:C++/Data_Structure_and_Algorithm/Sort/C/Selection_sort.c
#include <stdio.h>

// Function to swap two integers
void swap(int *xp, int *yp) {
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

// Function to perform selection sort
void selectionSort(int arr[], int n) {
    // Iterate over the array
    for (int i = 0; i < n; i++) {
        int mini = i; // Assume the current index has the minimum value
        // Iterate over the unsorted portion of the array
        for (int j = i; j < n; j++) {
            // Find the index of the minimum element in the unsorted portion
            if (arr[mini] > arr[j]) {
                mini = j;
            }
        }
        // If the minimum element is not at its original position, swap it with the current element
        if (mini != i) {
            swap(&arr[mini], &arr[i]);
        }
    }
}

int main() {
    int arr[] = {5, 4, 3, 2, 1}; // Input array
    int n = sizeof(arr) / sizeof(arr[0]); // Calculate the size of the array
    selectionSort(arr, n); // Call selection sort function
    // Print the sorted array
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
    return 0;
}